src/Features/LanguageServer/Protocol/Handler/RequestTelemetryLogger.cs

```diff
@@ -92,6 +94,14 @@ public void Dispose()
             return;
         }
 
+        // Flush all telemetry logged through TelemetryLogging
+        TelemetryLogging.Flush();
+
+        TelemetryLogging.Flushed -= OnFlushed;
+    }
+
+    private void OnFlushed(object? sender, EventArgs e)
+    {
         foreach (var kvp in _requestCounters)
         {
             TelemetryLogging.Log(FunctionId.LSP_RequestCounter, KeyValueLogMessage.Create(LogType.Trace, m =>
```

Comment:
I was trying to figure out why these didn't use `TelemetryLogging.LogAggregated` - but its because we're not using a bucket based aggregation here right?  We're just logging pure sums.

Wondering if we should have another variant of the LogAggregated that does a sum or something.  But maybe a change for a later date.

Reply:
Yes, that's exactly right. I considered that as part of this, but it seemed like overkill for now. Especially, as I know we have more upcoming work in this area as part of potentially moving towards OTel.